src/include/distributed/multi_copy.h

```diff
@@ -51,6 +52,15 @@ typedef struct NodeAddress
 	int32 nodePort;
 } NodeAddress;
 
+/* struct to allow rReceive to coerce tuples before sending them to workers  */
+typedef struct CopyCoercionData
+{
+	CoercionPathType coercionType;
+	FmgrInfo firstCoerceFunction;
+	FmgrInfo secondCoerceFunction;
```

Comment:
This could use better naming. Maybe:

```
/* function to use for direct coercions*/
FmgrInfo coerceFunction;

/* input and output functions for coercion via I/O */
FmgrInfo inputFunction;
FmgrInfo outputFunction;
```

I don't think the extra memory usage is really a concern, but alternative you could use pointers.

Reply:
Oh, you had the same idea. Yeah, I'll do this! An extra `FmgrInfo` per column isn't going to make a noticeable difference in memory usage.